server/bin/pytests/test_26.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+import os, sys
+import logging
+import pytest
+
+from pbench import PbenchConfig, BadConfig, get_pbench_logger
+
+_NAME_ = "pbench-logger-test"
+cfg_name = os.environ.get("CONFIG")
+logdir = "/var/tmp/pbench-test-server/test-26/pbench-local/logs"
+
+files = {
+    "file": "foo",
+    "devlog": "bar",
+    "hostport": "baz"
+    }
+
+messages = {
+    "file": "logger_type=file in file foo",
+    "devlog": "logger_type=devlog in file bar",
+    "hostport": "logger_type=hostport in file baz"
+}
+
+def mock_the_handler(logger, logger_type, fname):
+    
+    hdlr = logger.logger.handlers[0]
+    logger.logger.removeHandler(hdlr)
+
+    fh = logging.FileHandler(os.path.join(logdir,fname))
+    fh.setLevel(logging.DEBUG)
+    logger.logger.addHandler(fh)
+    
+    return logger
+
+def test_pbench_logger():
+
+    config = PbenchConfig(cfg_name)
+    logger = get_pbench_logger(_NAME_, config)
+
+    logger_type = config.get("logging", "logger_type")
+    log = mock_the_handler(logger, logger_type, files[logger_type])
+    log.debug(messages[logger_type])
+    
+    try:
+        if os.path.isfile(os.path.join(logdir,files[logger_type])):
+            with open(os.path.join(logdir,files[logger_type]), 'r') as f:
+                assert f.read()[:-1] == messages[logger_type]
+            os.remove(os.path.join(logdir,files[logger_type]))
```

Comment:
In fact, it would probably be a good idea to get the contents of the file into the output of the test, so we can compare it against the gold file.

Reply:
every time a different file is created as file, devlog, hostport have different filename, but gold file will only be able to see one file and will throw error for the other 2. We can do like, we can create all files but compare the content of only mentioned file, this can save the test.